Lib/fontTools/varLib/instancer.py

```diff
@@ -1123,6 +1123,28 @@ def updateNameTable(varfont, axisLimits):
 
     The updated name table will conform to the R/I/B/BI naming model.
     """
+    # This task can be split into two parts:
+
+    #   1. Collecting and sorting the relevant AxisValues:
+    # - First check the variable font has a STAT table and it contains
+    #   AxisValues.
+    # - Create a dictionary which contains the new default locations for each
+    #   axis.
+    # - Duplicate the existing stat table and instantiate it using the dict
+    #   we created in the last step. We should now have a new STAT table
+    #   which only contains AxisValues for the default axis locations.
```

Comment:
I think you should be able to get this list of AxisValues for the new default locations without having to make a duplicate STAT table and running `_instantiateSTAT` on it.

Reply:
you basically want to get the relevant axis values out from the STAT table, it's a matter of iterating over them and storing them in a list, no need for the copy deepcopy and re-running of instantiateSTAT. If you see there's logic duplicated between your loop and the instantiateSTAT itself, you can factor out the relevant portion and reuse accordingly

Reply:
> If you see there's logic duplicated between your loop and the instantiateSTAT itself, you can factor out the relevant portion and reuse accordingly

SGTM

Reply:
Yep, the implementation is now much cleaner. Thanks!